### ISSUE TITLE
[New] min-max scaling to improve Elastic Net regression model in parameter prediction of future phases

### ISSUE THREAD

**Issue by lisphilar** [lisphilar/covid19-sir#611]
## Summary of this new feature
Feature engineering is required to improve Elastic Net regression model in parameter prediction of future phases. `Scenario.fit()` will use feature engineering packages (e.g. `tsgresh` https://tsfresh.readthedocs.io/en/latest/) internally.

**Comment by lisphilar** [comment]
Min-max scaling is also required.
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html

Because explanatory variables will include cumulative values, normalization is not suitable.

**Comment by lisphilar** [comment]
The indicator used in `Scenario.estimate_delay()` (default: Stringency_index) should be removed from X in linear regression to avoid data leakage.

**Comment by lisphilar** [comment]
Output dictionary of `Scenario.fit()` will includes estimator name, transformer name, X/y datasets.

**Comment by lisphilar** [comment]
Becuase we can use `Vaccinations` etc. (values are changed every day), many phases will be created. For enhancement, only the values for the last date will be used.

**Comment by lisphilar** [comment]
Feature engineering was skipped because the number of records (parameters) are small (about 20 phases).

**Comment by lisphilar** [comment]
With #633, `Scenario.fit()` was improved as follows.
- Min-max scaling is introduced in the dataset for prediction.
- The column used for `Scenario.estimate_delay()` is removed from the dataset for prediction.
- The output dictionally includes the datasets and scaler/regressor classes.
